webapp/graphite/render/evaluator.py

```diff
@@ -1,17 +1,23 @@
 import re
+import sys
+from graphite.logger import log
 from graphite.render.grammar import grammar
 from graphite.render.datalib import fetchData, TimeSeries
+from graphite.exceptions.invalid_input_error import InvalidInputError
 
 
 def evaluateTarget(requestContext, target):
-  tokens = grammar.parseString(target)
-  result = evaluateTokens(requestContext, tokens)
+  try:
+    tokens = grammar.parseString(target)
+    result = evaluateTokens(requestContext, tokens)
 
-  if type(result) is TimeSeries:
-    return [result] #we have to return a list of TimeSeries objects
+    if type(result) is TimeSeries:
+      return [result] #we have to return a list of TimeSeries objects
 
-  else:
-    return result
+    else:
+      return result
+  except (TypeError, ValueError, KeyError, InvalidInputError) as err:
```

Comment:
do we have to catch InvalidInputError if you are throwing the same?

Reply:
Removed InvaidInputError from the list.